src/envoy/mixer/config.cc

```diff
@@ -93,8 +94,7 @@ bool ReadV2Config(const Json::Object& json, Message* message) {
     return false;
   }
   std::string v2_str = json.getObject(kV2Config)->asJsonString();
-  Status status =
-      ::google::protobuf::util::JsonStringToMessage(v2_str, message);
+  Status status = Utils::ParseJsonMessage(v2_str, message);
```

Comment:
I think we should just use `MessageUtil::loadFromJson` from envoy, the behavior should be consistent with it.

Reply:
That one is relying on a third-party lib to parse the json. I would rather ensure that the option is set correctly by using the protobuf lib directly, as the logic is simple here and shouldn't involve much overhead.

Reply:
No it doesn't use third-party lib to parse:
https://github.com/envoyproxy/envoy/blob/910058d7113e6439d077e2467e26abb7b803000e/source/common/protobuf/utility.cc#L38

Do we see any real problem with unknown fields? If you don't fix upstream, the issue could be still there.

Reply:
Then it could also be an issue for envoy. I will check the upstream.